src/Compilers/Core/Portable/RealParser.cs

```diff
@@ -29,21 +29,12 @@ internal static class RealParser
         /// <returns>True of the input was converted; false if there was an overflow</returns>
         public static bool TryParseDouble(string s, out double d)
         {
-            try
-            {
-                var str = DecimalFloatingPointString.FromSource(s);
-                var dbl = DoubleFloatingPointType.Instance;
-                ulong result;
-                var status = RealParser.ConvertDecimalToFloatingPointBits(str, dbl, out result);
-                d = BitConverter.Int64BitsToDouble((long)result);
-                return status != Status.Overflow;
-            }
-            catch (System.FormatException)
-            {
-                // this can occur when the exponent is empty (e.g. "0.0e") or too large to fit in an integer
-                d = 0.0;
-                return false;
-            }
+            var str = DecimalFloatingPointString.FromSource(s);
```

Comment:
Want to keep the `try / catch` here until we have more extensive testing on this code base.  The parsing not handling error cases sufficiently lead to the top reported issue on the compiler for Update 1 RC.  The `try / catch` is a good safeguard against that.